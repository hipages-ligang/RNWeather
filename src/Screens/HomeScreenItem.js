import React from 'react';
import { TouchableOpacity, View, Image, Text, StyleSheet } from 'react-native';
import Colors from '../Assets/Colors';
import { getTimeStr } from '../Utils/commonUtils';

const styles = StyleSheet.create({
    itemContainer: {
        height: 60,
        backgroundColor: Colors.cardBg,
        paddingHorizontal: 10,
        flexDirection: 'row',
        alignItems: 'center',
        marginHorizontal: 10,
        marginVertical: 5,
        borderWidth: 1,
        borderRadius: 5,
        borderColor: Colors.border
    },
    itemWeatherIcon: {
        width: 50,
        height: 50,
        backgroundColor: Colors.primary_light,
        marginRight: 10,
        borderRadius: 3
    }
});

export default HomeScreenItem = ({ item, onPressCallback }) => {
    let weather = item?.weather[0];
    let maxTemp = item?.temp?.max;
    let minTemp = item?.temp?.min;
    let timeStr = getTimeStr(item?.dt);

    return (
        <TouchableOpacity
            onPress={() => {
                if (onPressCallback) {
                    onPressCallback();
                }
            }}
        >
            <View style={styles.itemContainer} key={timeStr}>
                <Image
                    style={styles.itemWeatherIcon}
                    source={{
                        uri: `https://openweathermap.org/img/wn/${weather.icon}@2x.png`
                    }}
                />
                <View>
                    <Text>{weather.main}</Text>
                    <Text>{`${parseInt(maxTemp)} / ${parseInt(
                        minTemp
                    )} â„ƒ`}</Text>
                </View>
                <View style={{ flex: 1, alignItems: 'flex-end' }}>
                    <Text>{timeStr}</Text>
                </View>
            </View>
        </TouchableOpacity>
    );
};
