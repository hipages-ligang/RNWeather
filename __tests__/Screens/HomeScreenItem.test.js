import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreenItem from '../../src/Screens/HomeScreenItem';

describe('<HomeScreenItem />', () => {
    const props = {
        item: {
            dt: 1633665600,
            sunrise: 1633646640,
            sunset: 1633688411,
            moonrise: 1633652820,
            moonset: 1633693200,
            moon_phase: 0.06,
            temp: {
                day: 16.81,
                min: 13.88,
                max: 17.35,
                night: 15.53,
                eve: 16.09,
                morn: 14.02
            },
            feels_like: { day: 16.5, night: 15.28, eve: 15.87, morn: 13.78 },
            pressure: 1016,
            humidity: 75,
            dew_point: 11.77,
            wind_speed: 2.64,
            wind_deg: 41,
            wind_gust: 4.99,
            weather: [
                {
                    id: 804,
                    main: 'Clouds',
                    description: 'overcast clouds',
                    icon: '04d'
                }
            ],
            clouds: 96,
            pop: 0.42,
            uvi: 0.94
        }
    };
    it('home screen item snapshot', () => {
        const homeScreenTree = renderer
            .create(<HomeScreenItem {...props} />)
            .toJSON();
        expect(homeScreenTree).toMatchSnapshot();
    });
});
