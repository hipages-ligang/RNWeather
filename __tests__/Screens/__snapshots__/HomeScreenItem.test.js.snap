// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreenItem /> home screen item snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eeeeee",
        "borderColor": "#cccccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 60,
        "marginHorizontal": 10,
        "marginVertical": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://openweathermap.org/img/wn/04d@2x.png",
        }
      }
      style={
        Object {
          "backgroundColor": "#5fceea",
          "borderRadius": 3,
          "height": 50,
          "marginRight": 10,
          "width": 50,
        }
      }
    />
    <View>
      <Text>
        Clouds
      </Text>
      <Text>
        17 / 13 â„ƒ
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <Text>
        Fri 8/10
      </Text>
    </View>
  </View>
</View>
`;
